ipfs-cluster-service/main.go

```diff
@@ -261,7 +261,82 @@ removal, upgrade state using this command, and restart every peer.
 					Action: func(c *cli.Context) error {
 						err := upgrade()
 						checkErr("upgrading state", err)
-						return err
+						return nil
+					},
+				},
+				{
+					Name:  "export",
+					Usage: "save the IPFS Cluster state to a human readable json file",
+					Description: `
+
+This command reads the current cluster state and saves it as json for 
+human readability and editing.  In general any version of cluster will only
```

Comment:
Let's not say "in general", either things are going to be migrated or not (edit `FromJson` does not seem to allow for any migrations).

If  my recommendations above are taken, we can get rid of Restore (and FromJSON). As long as `api` types don't change in an incompatible way, we should be able parse to re-import states via calls to Add(). And otherwise it's always easy to parse some JSON and transform it with a helper tool. 

It's not a crazy requirement to say that exported states should only be imported in the same cluster version, and then use the regular `upgrade` path to upgrade.

 We can revisit upgrading paths as part of #230 anyways.

Reply:
I am on board.  One implication of getting rid of Restore (Migrate) in this case is that `export` must be called with a version of `ipfs-cluster-service` with the same state representation as the snapshot being exported.  Unmarshal won't do anything besides version parsing on other state formats.  I am fine with this assumption, just making it explicit.

My wording here was confusing but your suggested assumption that exported states should only be imported in the same cluster version was exactly what I was trying to communicate.

Reply:
> One implication of getting rid of Restore (Migrate) in this case is that export must be called with a version of ipfs-cluster-service with the same state representation as the snapshot being exported

ok, we can do the version check just like when cluster is started and warn the user if it doesn't match right?

> Unmarshal won't do anything besides version parsing on other state formats

I'm not sure what you mean here

Reply:
I understand now. Yeah, maybe using Restore() directly is the best, as it will do unmarshaling or migration as needed, but will get something done (and as pointed out in the last round, code can be shared with upgrade()).